utils/sub8_ros_tools/sub8_ros_tools/image_helpers.py

```diff
@@ -66,8 +66,13 @@ def __init__(self, topic, callback=None, encoding="bgr8", queue_size=1):
         Assumes topic of type "sensor_msgs/Image"
         This behaves like a conventional subscriber, except handling the additional image conversion
         '''
+        if callback is None:
+            callback = lambda im: setattr(self, 'last_image', im)
+
         self.encoding = encoding
         self.camera_info = None
+        self.last_image_time = None
```

Comment:
It'd be a good idea to refactor `follow_orange_pipes` to use this timestamp instead of its own.